### ISSUE TITLE
ESLint plugin doesn't recognize typescript-eslint's stylistic(-type-checked) or strict-type-checked rulesets

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [webpro/knip#154]
Coming over from https://github.com/JoshuaKGoldberg/template-typescript-node-package/pull/601, my ESLint config contains these plugins (among others) under `extends`:

```plaintext
"plugin:@typescript-eslint/strict",
"plugin:@typescript-eslint/stylistic",
"plugin:@typescript-eslint/strict-type-checked",
"plugin:@typescript-eslint/stylistic-type-checked",
```

...and Knip only recognizes `strict` from them:

```plaintext
 $ pnpm run lint:knip

> template-typescript-node-package@1.28.40 lint:knip /Users/josh/repos/template-typescript-node-package
> knip

Unlisted dependencies (3)
@typescript-eslint/eslint-plugin-strict-type-checked     .eslintrc.cjs
@typescript-eslint/eslint-plugin-stylistic               .eslintrc.cjs
@typescript-eslint/eslint-plugin-stylistic-type-checked  .eslintrc.cjs
 ELIFECYCLE  Command failed with exit code 3.
```

**Comment by JoshuaKGoldberg** [comment]
Fun fact - if you add `stylistic.*|strict.*` to the regex in https://github.com/webpro/knip/blob/d3e6f01f6bf2bd437e774c1d615c3c2bc5c56032/src/plugins/eslint/helpers.ts#L77, it works right. Would you be up for this as a bandaid fix pending the big _`TODO`_ comment in that file?

https://github.com/webpro/knip/blob/d3e6f01f6bf2bd437e774c1d615c3c2bc5c56032/src/plugins/eslint/helpers.ts#L69-L71

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.15.2. See [Release 2.15.2](https://github.com/webpro/knip/releases/tag/2.15.2) for release notes._

**Comment by webpro** [comment]
Thanks for reporting this, and thanks for the work in `@typescript-eslint/*` v6!

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.15.3. See [Release 2.15.3](https://github.com/webpro/knip/releases/tag/2.15.3) for release notes._
